source/patterns/@aws-solutions-constructs/core/lib/cloudfront-distribution-helper.ts

```diff
@@ -178,3 +183,74 @@ export function CloudFrontDistributionForS3(scope: cdk.Construct,
     };
     return [cfDistribution, edgeLambdaVersion, loggingBucket];
 }
+
+export function CloudFrontDistributionForMediaStore(scope: cdk.Construct,
+                                                    mediaStoreContainer: mediastore.CfnContainer,
+                                                    cloudFrontDistributionProps?: cloudfront.DistributionProps | any):
+                                                    [cloudfront.Distribution, s3.Bucket, cloudfront.OriginRequestPolicy] {
+
+    let defaultprops: cloudfront.DistributionProps;
+    let originRequestPolicy: cloudfront.OriginRequestPolicy;
+    let loggingBucket: s3.Bucket;
+
+    if (cloudFrontDistributionProps && cloudFrontDistributionProps.enableLogging && cloudFrontDistributionProps.logBucket) {
+        loggingBucket = cloudFrontDistributionProps.logBucket as s3.Bucket;
+    } else {
+        loggingBucket = createLoggingBucket(scope, 'CloudfrontLoggingBucket');
+    }
+
+    if (cloudFrontDistributionProps
+        && cloudFrontDistributionProps.defaultBehavior
+        && cloudFrontDistributionProps.defaultBehavior.originRequestPolicy) {
+        originRequestPolicy = cloudFrontDistributionProps.defaultBehavior.originRequestPolicy;
+    } else {
+        const originRequestPolicyProps: cloudfront.OriginRequestPolicyProps = {
```

Comment:
Is there any reference on how did you come up with this origin request policy? Wonder if it makes sense to apply for the other aws-cloudfront-* constructs as well ?

Reply:
Oh, for the existing solution we have is using the legacy setting for the default cache behavior, but CDK doesn't support the legacy way. To mitigate it, I've put the origin request policy instead.